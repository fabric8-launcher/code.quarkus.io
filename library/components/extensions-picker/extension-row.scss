@import "../config.scss";

#extension-origin-platform-tooltip > .tooltip-inner {
  max-width: 400px;
}

.extension-row {
  display: flex;
  width: 100%;
  line-height: 30px;
  cursor: pointer;
  flex-wrap: wrap;

  svg {
    pointer-events: none;
  }

  .extension-selector {
    color: var(--extensionsPickerCheckColor);
  }

  &.selected .extension-selector {
    color: var(--extensionsPickerCheckSelectedColor);
  }

  &.hover.selected .extension-selector {
    color: var(--extensionsPickerCheckSelectedHoverColor);
  }

  div:focus {
    outline: none;
  }

  .extension-remove {
    color: var(--extensionsPickerRemoveButtonColor);
    height: 30px;
  }

  .extension-selector {

    @media screen and (max-width: $mobileWidth) {
      display: none;
    }
  }

  .extension-summary {
    overflow: hidden;
    display: flex;
  }

  .extension-origin-platform {
    font-size: 0.8rem;
    color: #84C590;
    margin-top: 2px;
    font-weight: normal;
  }

  .extension-name {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    font-size: 0.8rem;
    font-weight: bold;
    height: 24px;
    outline: none;
  }

  .extension-id {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    font-size: 0.8rem;
    margin: 0 5px;
    height: 24px;
    outline: none;
    color: var(--extensionsPickerIdTextColor);
    @media screen and (max-width: $mobileWidth) {
      display: none;
    }
  }

  .extension-tag {
    font-size: 0.6rem;
    line-height: 20px;
    margin: 5px 5px;
    padding: 0 3px;
    height: 20px;
    white-space: nowrap;
    text-transform: uppercase;
    border: 1px solid black;
  }

  .extension-tag-dropdown .dropdown-menu {
    padding: 5px;
    font-size: 0.8rem;
    max-width: 500px;
  }
  a.extension-tag {
    cursor: pointer;
  }

  .extension-description {
    padding-left: 30px;
    padding-right: 30px;
    font-size: 0.8rem;
    color: var(--extensionsPickerDescriptionTextColorOnTablet);
  }

  .extension-more {
    max-width: 150px;
    font-size: 0.8rem;
    height: 33px;

    .extension-more-button {
      background-color: transparent;
      border: none;
      color: var(--extensionsPickerTextColor);
      outline: none;
      padding: 0.5rem;
      font-size: 1rem;
      cursor: pointer;

      display: flex;
      font-weight: 400;
      align-items: center;
      justify-content: space-between;
    }
    .dropdown-item {
      width: 330px;
      font-weight: normal;
    }
    .dropdown-item .copy-to-clipboard {
      display: flex;
      width: 100%;
    }
  }
}