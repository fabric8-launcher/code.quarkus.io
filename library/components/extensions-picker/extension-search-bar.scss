.search-bar {
  margin: 0 auto;
  width: 100%;
  display: flex;

  .form-group {
    border: 1px solid var(--extensionsPickerSearchBorderColor);
    border-left: none;
    display: flex;
    flex-grow: 1;

    svg.search-icon {
      margin: 10px;
      width: 20px;
      font-size: 1.2rem;
    }
    &:focus-within {
      border-color: var(--extensionsPickerSearchBorderColorOnFocus);

      svg {
        color: var(--extensionsPickerSearchBorderColorOnFocus);
      }
    }
  }
  .filter-shortcut {
    .dropdown-item.list-item {
      width: 700px;
      height: auto;
      display: block;
      border: none;
      cursor: auto;
      line-height: 30px;

      span {
        display: inline-block;
        margin-left: 7px;
        font-size: 0.9rem;
        cursor: pointer;
        color: var(--extensionsPickerSearchBorderColorOnFocus);
        border: none;
        text-decoration: underline;
      }

      span.origin {

        &.current-origin {
          font-weight: bold;
        }
      }
    }
    .dropdown-item {
      font-weight: normal;
      line-height: 35px;
      height: 35px;
    }
  }

  .filter-shortcut-button {
    display: flex;
    cursor: pointer;
    background-color: transparent;
    color: var(--extensionsPickerSearchTextColor);
    line-height: 40px;
    font-weight: bold;
    padding: 0 10px;
    align-items: center;
    border: 1px solid var(--extensionsPickerSearchBorderColor);
    svg {
      font-size: 1rem;
      width: 20px;
      margin-left: 10px;
    }
  }

  input.form-control {
    font-size: 14px;
    height: 40px;
    line-height: 40px;
    width: 100%;
    background: transparent;
    color: var(--extensionsPickerSearchTextColor);
    font-family: var(--inputFont);
    border: none;
    outline: none;

    &:focus {
      outline: none;
    }
  }
}

