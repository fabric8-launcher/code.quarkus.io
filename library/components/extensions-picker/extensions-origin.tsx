import React from 'react';
import { OverlayTrigger, Tooltip } from 'react-bootstrap';
import { FaAtom } from 'react-icons/fa';

export function ExtensionsOrigin(props: { platform: boolean }) {
  const Overlay = (
    <Tooltip id="extension-origin-platform-tooltip" style={{ zIndex: 200 }}>
      The atom indicates the extension is part of the selected Quarkus Platform. Using extensions from the platform are tested and verified together and thus what is default recommended as easier to use and upgrade.
    </Tooltip>
  );
  return (
    props.platform ? <OverlayTrigger
      placement="bottom"
      overlay={Overlay}
      delay={{ show: 200, hide: 0 }}
    >
      <span className="extension-origin-platform">
        <FaAtom />
      </span>
    </OverlayTrigger> : <React.Fragment/>
  )
}