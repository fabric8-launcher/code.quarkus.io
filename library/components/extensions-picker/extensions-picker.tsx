import React, { SetStateAction, useEffect, useRef, useState } from 'react';
import { useHotkeys } from 'react-hotkeys-hook';
import { useAnalytics } from '@quarkusio/code-quarkus.core.analytics';
import { InputProps } from '@quarkusio/code-quarkus.core.types';
import {
  debouncedComputeResults,
  FilterResult, ProcessedExtensions,
  processExtensionsValues
} from './extensions-utils';
import { QuarkusProject } from '../api/model';
import './extensions-picker.scss';
import { ExtensionRow } from './extension-row';
import { ExtensionSearchBar } from './extension-search-bar';
import { Button } from 'react-bootstrap';
import { FaCaretDown } from 'react-icons/fa';
import { SearchResultsInfo } from './search-results-info';

export interface ExtensionEntry {
  id: string;
  name: string;
  version: string;
  keywords: string[];
  tags: string[];
  description?: string;
  shortName?: string;
  category: string;
  order: number;
  default: boolean;
  guide?: string;
  bom?: string
  platform: boolean;
}

export interface TagEntry {
  name: string;
  href?: string;
  description?: string;
  color?: string;
  hide?: boolean;
}

export interface ExtensionsPickerValue {
  extensions: ExtensionEntry[];
}

const REDUCED_SIZE = 100;

interface ExtensionsPickerProps extends InputProps<ExtensionsPickerValue> {
  entries: ExtensionEntry[];
  tagsDef: TagEntry[];
  placeholder: string;
  buildTool: string;
  project?: QuarkusProject;

  filter: string;
  setFilter: React.Dispatch<SetStateAction<string>>;

  filterFunction?(d: ExtensionEntry): boolean;
}

const hotkeysOptions = {
  filter: (e) => {
    const el = (e.target) as any | undefined;
    if (!el) {
      return true;
    }
    const tagName = el && el.tagName;
    return el.id === 'extensions-search-input' || !(tagName === 'INPUT' || tagName === 'SELECT' || tagName === 'TEXTAREA');
  },
  enableOnTags: [ 'INPUT' ] as 'INPUT'[]
};

export const ExtensionsPicker = (props: ExtensionsPickerProps) => {
  const { filter } = props;
  const [ processedExtensions, setProcessedExtensions ] = useState<ProcessedExtensions | undefined>(undefined);
  const [ keyboardActivated, setKeyBoardActivated ] = useState<number>(-1);
  const [ showAll, setShowAll ] = useState<boolean>(false);
  const [ result, setResult ] = useState<FilterResult | undefined>();
  const analytics = useAnalytics();
  const context = { element: 'extension-picker' };

  function setFilter(filter: string) {
    setKeyBoardActivated(-1);
    setShowAll(false);
    props.setFilter(filter);
  }
  const extensions = props.value.extensions || [];

  const entrySet = new Set(extensions.map(e => e.id));
  const entriesById: Map<string, ExtensionEntry> = new Map(props.entries.map(item => [ item.id, item ]));

  useEffect(() => {
    setProcessedExtensions(processExtensionsValues(props.entries));
  }, [ props.entries, setProcessedExtensions ]);
  
  useEffect(() => {
    debouncedComputeResults(analytics, filter, props.entries, processedExtensions, setResult);
  }, [ filter, processedExtensions, props.entries, setShowAll, setResult ]);

  const allEntries = result?.effective || [];
  const entries = showAll ? allEntries : allEntries.slice(0, REDUCED_SIZE)
  const add = (index: number, type: string) => {
    const id =entries[index].id;
    entrySet.add(id);
    props.onChange({ extensions: Array.from(entrySet).map(e => entriesById.get(e)!) });
    analytics.event('Select extension', { extension: id, type, ...context });
    if (keyboardActivated >= 0) {
      setKeyBoardActivated(index);
    }
  };

  const remove = (id: string, type: string) => {
    entrySet.delete(id);
    props.onChange({ extensions: Array.from(entrySet).map(e => entriesById.get(e)!) });
    analytics.event('Unselect extension', { extension: id, type, ...context });
  };

  const flip = (index: number, origin: string) => {
    if (!entries[index] || entries[index].default) {
      return;
    }
    if (entrySet.has(entries[index].id)) {
      remove(entries[index].id, origin);
    } else {
      add(index, origin);
    }
  };


  useHotkeys('esc', () => setKeyBoardActivated(-1), hotkeysOptions);
  useHotkeys('up', () => setKeyBoardActivated((prev) => Math.max(0, prev - 1)), hotkeysOptions);
  useHotkeys('down', () => setKeyBoardActivated((prev) => Math.min(entries.length - 1, prev + 1)), hotkeysOptions, [ entries ]);
  useHotkeys('space', (event) => {
    if (keyboardActivated >= 0) {
      event.preventDefault();
      flip(keyboardActivated, 'Keyboard');
    }
  }, hotkeysOptions, [ entries, keyboardActivated ]);

  let currentCat: string | undefined;

  return (
    <div className="extensions-picker" aria-label="Extensions picker">
      <div className="control-container">
        <ExtensionSearchBar placeholder={props.placeholder} filter={filter} project={props.project}
          setFilter={setFilter} result={result}/>
      </div>
      <div className="main-container responsive-container">
        <SearchResultsInfo filter={props.filter} setFilter={props.setFilter} result={result} />
        <div className="extension-list-wrapper">
          {entries.map((ex, i) => {
            const ext = (
              <ExtensionRow
                selected={entrySet.has(ex.id)}
                keyboardActived={i === keyboardActivated}
                {...ex}
                key={i}
                tagsDef={props.tagsDef}
                onClick={() => flip(i, 'List')}
                buildTool={props.buildTool}
                pickerLayout={true}
              />
            );
            if (!result.filtered && (!currentCat || currentCat !== ex.category)) {
              currentCat = ex.category;
              return (
                <React.Fragment key={i}>
                  <div className="extension-category">
                    {currentCat}
                  </div>
                  {ext}
                </React.Fragment>
              );
            }
            return ext;
          })}
        </div>
        {!showAll && allEntries.length > 100 && <Button className='button-show-more btn-base' onClick={() => setShowAll(true)}><FaCaretDown /> Show more extensions</Button>}
      </div>
    </div>
  );
};

