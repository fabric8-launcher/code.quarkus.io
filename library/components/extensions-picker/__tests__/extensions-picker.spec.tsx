import * as React from 'react'
import { cleanup, fireEvent, render } from '@testing-library/react';
import { ExtensionEntry, ExtensionsPicker } from '../extensions-picker';
import { act } from 'react-dom/test-utils';
import { DEFAULT_TAGS } from '../../api/code-quarkus-api';

afterEach(() => {
  cleanup();
});

const entries: ExtensionEntry[] = [
  {
    'id': 'io.quarkus:quarkus-arc',
    'name': 'ArC',
    'version': 'test-version',
    'tags': [],
    'keywords': [
      'arc',
      'cdi',
      'dependency-injection',
      'di',
      'label'
    ],
    'platform': true,
    'default': false,
    'description': 'Build time CDI dependency injection',
    'shortName': 'CDI',
    'category': 'Core',
    'order': 0,
  },
  {
    'id': 'io.quarkus:quarkus-camel-netty4-http',
    'version': 'test-version',
    'name': 'Camel Netty4 test HTTP',
    'tags': [ 'status:preview' ],
    'default': false,
    'keywords': [
      'camel-netty4-http',
      'camel'
    ],
    'platform': true,
    'description': 'Camel support for Netty',
    'category': 'Integration',
    'order': 2,
  },
  {
    'id': 'some-id',
    'version': 'test-version',
    'name': 'A CDI in name test',
    'tags': [ 'status:experimental' ],
    'default': false,
    'keywords': [
      'lambda',
      'amazon-lambda',
      'aws-lambda',
      'amazon',
      'aws',
      'label'
    ],
    'platform': true,
    'shortName': 'a shortname',
    'description': 'Some description',
    'category': 'Cloud',
    'order': 1,
  },
];

describe('<ExtensionsPicker />', () => {

  it('renders the ExtensionsPicker correctly', () => {
    const setFilter = jest.fn();
    const comp = render(<ExtensionsPicker placeholder="" entries={entries} value={{ extensions: [] }} onChange={() => { }} buildTool="MAVEN" filter="" setFilter={setFilter} tagsDef={DEFAULT_TAGS} />);
    expect(comp.asFragment()).toMatchSnapshot();
  });

  it('show results for valid search', async () => {
    const setFilter = jest.fn();
    const comp = render(<ExtensionsPicker placeholder="" entries={entries} value={{ extensions: [] }} onChange={() => { }} buildTool="MAVEN" filter="" setFilter={setFilter} tagsDef={DEFAULT_TAGS} />);

    const searchField = comp.getByLabelText('Search extensions');
    fireEvent.change(searchField, { target: { value: 'CDI' } });
    expect(setFilter).toBeCalledTimes(1);
    expect(setFilter).lastCalledWith('CDI');
    act(() => {
      comp.rerender(<ExtensionsPicker placeholder="" entries={entries} value={{ extensions: [] }} onChange={() => { }} buildTool="MAVEN" filter="CDI" setFilter={setFilter} tagsDef={DEFAULT_TAGS} />);
    });
    const result = await comp.findAllByText(entries[0].description!);
    expect((result as HTMLElement[]).length).toBe(1);
    expect(comp.asFragment()).toMatchSnapshot();
  });

  it('select values and save', async () => {
    const setFilter = jest.fn();
    const handleChange = jest.fn();
    const comp = render(<ExtensionsPicker placeholder="" entries={entries} value={{ extensions: [] }} onChange={handleChange} buildTool="MAVEN" filter="" setFilter={setFilter} tagsDef={DEFAULT_TAGS} />);

    const searchField = comp.getByLabelText('Search extensions');
    fireEvent.change(searchField, { target: { value: 'netty' } });
    expect(setFilter).toBeCalledTimes(1);
    expect(setFilter).lastCalledWith('netty');
    act(() => {
      comp.rerender(<ExtensionsPicker placeholder="" entries={entries} value={{ extensions: [] }} onChange={() => { }} buildTool="MAVEN" filter="netty" setFilter={setFilter} tagsDef={DEFAULT_TAGS} />);
    });
    const item = await comp.findAllByText(entries[1].description!);
    fireEvent.click(item[0]);
    expect(comp.asFragment()).toMatchSnapshot();
  });
});
