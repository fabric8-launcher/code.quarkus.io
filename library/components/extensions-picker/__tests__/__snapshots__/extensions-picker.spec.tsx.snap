// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtensionsPicker /> renders the ExtensionsPicker correctly 1`] = `
<DocumentFragment>
  <div
    aria-label="Extensions picker"
    class="extensions-picker"
  >
    <div
      class="control-container"
    >
      <div
        class="search-bar responsive-container"
      >
        <div
          class="filter-shortcut dropdown"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="filter-shortcut-button dropdown-toggle btn btn-primary"
            type="button"
          >
            Filters 
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              />
            </svg>
          </button>
        </div>
        <div
          class="form-group"
        >
          <svg
            class="search-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <input
            aria-label="Search extensions"
            autocomplete="off"
            class="extensions-search-input form-control"
            id="extensions-search-input"
            placeholder=""
            type="search"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="main-container responsive-container"
    >
      <div
        class="extension-list-wrapper"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ExtensionsPicker /> select values and save 1`] = `
<DocumentFragment>
  <div
    aria-label="Extensions picker"
    class="extensions-picker"
  >
    <div
      class="control-container"
    >
      <div
        class="search-bar responsive-container"
      >
        <div
          class="filter-shortcut dropdown"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="filter-shortcut-button dropdown-toggle btn btn-primary"
            type="button"
          >
            Filters 
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              />
            </svg>
          </button>
        </div>
        <div
          class="form-group"
        >
          <svg
            class="search-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <input
            aria-label="Search extensions"
            autocomplete="off"
            class="extensions-search-input form-control"
            id="extensions-search-input"
            placeholder=""
            type="search"
            value="netty"
          />
        </div>
      </div>
    </div>
    <div
      class="main-container responsive-container"
    >
      <div
        class="search-results-info"
      >
        <span
          class="origins-count"
        >
          <span
            class="results"
          >
            Extensions found by origin: 
          </span>
          <span
            class="origin-count platform-origin"
          >
            <span
              class="count"
            >
              1
            </span>
             in platform
          </span>
          <span
            class="origin-count other-origin"
          >
            <span
              class="count"
            >
              0
            </span>
             in other
          </span>
        </span>
        <a
          class="clear-button btn btn-primary"
        >
          Clear filters
        </a>
      </div>
      <div
        class="extension-list-wrapper"
      >
        <div
          class="extension-row"
        >
          <div
            aria-label="Switch io.quarkus:quarkus-camel-netty4-http extension"
            class="extension-selector"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z"
              />
            </svg>
          </div>
          <div
            class="extension-summary"
          >
            <span
              class="extension-name"
              title="Camel Netty4 test HTTP (test-version)"
            >
              Camel Netty4 test HTTP
            </span>
            <span
              class="extension-id"
              title="io.quarkus:quarkus-camel-netty4-http"
            >
               [quarkus-camel-netty4-http]
            </span>
            <span
              class="extension-origin-platform"
            >
              <svg
                enable-background="new 0 0 32 32"
                height="1em"
                viewBox="0 0 32 32"
                width="1em"
                xml:space="preserve"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="platform-icon_svg__st0"
                  d="m16 4.1 10.3 6V22L16 28 5.7 22V10L16 4.1M16 0 2.1 8v16L16 32l13.9-8V8L16 0z"
                />
                <path
                  class="platform-icon_svg__st0"
                  d="M23.8 11.5c-.5-.8-1.6-1.1-2.4-.7L16 14l-5.4-3.1c-.8-.5-1.9-.2-2.4.6s-.2 1.9.6 2.4l5.4 3.1v6.2c0 1 .8 1.8 1.7 1.8 1 0 1.8-.8 1.8-1.7v-6.2l5.4-3.1c.9-.6 1.2-1.6.7-2.5z"
                />
              </svg>
            </span>
            <div
              class="extension-tag-dropdown dropdown"
              delay="[object Object]"
              placement="right"
            >
              <div
                aria-expanded="false"
                aria-haspopup="true"
                class="extension-tag status:preview dropdown-toggle"
                style="border-color: #4695eb;"
              >
                <span>
                  PREVIEW
                </span>
              </div>
            </div>
          </div>
          <div
            class="extension-description"
            title="Camel support for Netty"
          >
            Camel support for Netty
          </div>
          <div
            class="extension-more"
          >
            <div
              class="dropdown"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="extension-more-button dropdown-toggle btn btn-primary"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 320 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ExtensionsPicker /> show results for valid search 1`] = `
<DocumentFragment>
  <div
    aria-label="Extensions picker"
    class="extensions-picker"
  >
    <div
      class="control-container"
    >
      <div
        class="search-bar responsive-container"
      >
        <div
          class="filter-shortcut dropdown"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            class="filter-shortcut-button dropdown-toggle btn btn-primary"
            type="button"
          >
            Filters 
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 320 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              />
            </svg>
          </button>
        </div>
        <div
          class="form-group"
        >
          <svg
            class="search-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
          <input
            aria-label="Search extensions"
            autocomplete="off"
            class="extensions-search-input form-control"
            id="extensions-search-input"
            placeholder=""
            type="search"
            value="CDI"
          />
        </div>
      </div>
    </div>
    <div
      class="main-container responsive-container"
    >
      <div
        class="search-results-info"
      >
        <span
          class="origins-count"
        >
          <span
            class="results"
          >
            Extensions found by origin: 
          </span>
          <span
            class="origin-count platform-origin"
          >
            <span
              class="count"
            >
              2
            </span>
             in platform
          </span>
          <span
            class="origin-count other-origin"
          >
            <span
              class="count"
            >
              0
            </span>
             in other
          </span>
        </span>
        <a
          class="clear-button btn btn-primary"
        >
          Clear filters
        </a>
      </div>
      <div
        class="extension-list-wrapper"
      >
        <div
          class="extension-row"
        >
          <div
            aria-label="Switch io.quarkus:quarkus-arc extension"
            class="extension-selector"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z"
              />
            </svg>
          </div>
          <div
            class="extension-summary"
          >
            <span
              class="extension-name"
              title="ArC (test-version)"
            >
              ArC
            </span>
            <span
              class="extension-id"
              title="io.quarkus:quarkus-arc"
            >
               [quarkus-arc]
            </span>
            <span
              class="extension-origin-platform"
            >
              <svg
                enable-background="new 0 0 32 32"
                height="1em"
                viewBox="0 0 32 32"
                width="1em"
                xml:space="preserve"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="platform-icon_svg__st0"
                  d="m16 4.1 10.3 6V22L16 28 5.7 22V10L16 4.1M16 0 2.1 8v16L16 32l13.9-8V8L16 0z"
                />
                <path
                  class="platform-icon_svg__st0"
                  d="M23.8 11.5c-.5-.8-1.6-1.1-2.4-.7L16 14l-5.4-3.1c-.8-.5-1.9-.2-2.4.6s-.2 1.9.6 2.4l5.4 3.1v6.2c0 1 .8 1.8 1.7 1.8 1 0 1.8-.8 1.8-1.7v-6.2l5.4-3.1c.9-.6 1.2-1.6.7-2.5z"
                />
              </svg>
            </span>
          </div>
          <div
            class="extension-description"
            title="Build time CDI dependency injection"
          >
            Build time CDI dependency injection
          </div>
          <div
            class="extension-more"
          >
            <div
              class="dropdown"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="extension-more-button dropdown-toggle btn btn-primary"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 320 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
        <div
          class="extension-row"
        >
          <div
            aria-label="Switch some-id extension"
            class="extension-selector"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zm-6 400H54c-3.3 0-6-2.7-6-6V86c0-3.3 2.7-6 6-6h340c3.3 0 6 2.7 6 6v340c0 3.3-2.7 6-6 6z"
              />
            </svg>
          </div>
          <div
            class="extension-summary"
          >
            <span
              class="extension-name"
              title="A CDI in name test (test-version)"
            >
              A CDI in name test
            </span>
            <span
              class="extension-id"
              title="some-id"
            >
               []
            </span>
            <span
              class="extension-origin-platform"
            >
              <svg
                enable-background="new 0 0 32 32"
                height="1em"
                viewBox="0 0 32 32"
                width="1em"
                xml:space="preserve"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class="platform-icon_svg__st0"
                  d="m16 4.1 10.3 6V22L16 28 5.7 22V10L16 4.1M16 0 2.1 8v16L16 32l13.9-8V8L16 0z"
                />
                <path
                  class="platform-icon_svg__st0"
                  d="M23.8 11.5c-.5-.8-1.6-1.1-2.4-.7L16 14l-5.4-3.1c-.8-.5-1.9-.2-2.4.6s-.2 1.9.6 2.4l5.4 3.1v6.2c0 1 .8 1.8 1.7 1.8 1 0 1.8-.8 1.8-1.7v-6.2l5.4-3.1c.9-.6 1.2-1.6.7-2.5z"
                />
              </svg>
            </span>
            <div
              class="extension-tag-dropdown dropdown"
              delay="[object Object]"
              placement="right"
            >
              <div
                aria-expanded="false"
                aria-haspopup="true"
                class="extension-tag status:experimental dropdown-toggle"
                style="border-color: #ff004a;"
              >
                <span>
                  EXPERIMENTAL
                </span>
              </div>
            </div>
          </div>
          <div
            class="extension-description"
            title="Some description"
          >
            Some description
          </div>
          <div
            class="extension-more"
          >
            <div
              class="dropdown"
            >
              <button
                aria-expanded="false"
                aria-haspopup="true"
                class="extension-more-button dropdown-toggle btn btn-primary"
                type="button"
              >
                <svg
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 320 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
